ApiClient/Source/Calendar/iCalendarParser/Parser.swift

```diff
@@ -0,0 +1,139 @@
+    static let eventEnd = "END:VEVENT"
+}
+
+public enum ParseError: Error{
+    case networkError(String)
+}
+
+/**
+ This class is used to parse `iCalendar` files.
+ 
+ - Warning:
+ The parser has only been tested with SchedJoules generated `iCalendar` files. It is not guranteed to work on files with even slightly different formatting.
+ */
+public final class Parser {
+    /// Parse the parameters into a dictionary
+    private static func parse(params: [String]) -> [String:String] {
+        var paramsDict = [String:String]()
+        for param in params{
+            let paramExploded = param.components(separatedBy: "=")
+            paramsDict[paramExploded[0]] = paramExploded[1]
+        }
+        return paramsDict
+    }
+    
+    /// Parse a single line of the ICS file
+    private static func parse(line: String) -> ParsedLine {
+        let firstPart = line.components(separatedBy: ":")[0]
+        let value = String(line.dropFirst(firstPart.count+1))
+        
+        if firstPart.contains(";"){
+            let key = firstPart.components(separatedBy: ";")[0]
+            let paramsStripped = firstPart.dropFirst(key.count+1).components(separatedBy: ";")
+            let params = parse(params: paramsStripped)
+            let parsedLine = ParsedLine(key: key, value: value, params: params)
+            return parsedLine
+        } else {
+            let key = firstPart
+            let parsedLine = ParsedLine(key: key, value: value, params: nil)
+            return parsedLine
+        }
+    }
+    
+    /// Parse the lines of the ICS file
+    private static func parse(lines: [String]) -> [ParsedLine] {
+        var parsedLines = [ParsedLine]()
+        for line in lines{
+            let parsedLine = parse(line: line)
+            parsedLines.append(parsedLine)
+        }
+        return parsedLines
+    }
+    
+    /// Parse ICS file
+    public static func parse(ics: String) -> Calendar {
+        // Events to return
+        var events = [Event]()
+        
+        // Keep track of parsing
+        var inEvent = false
+        
+        // Break up the ics into seperate lines
+        let lines = ics.components(separatedBy: CharacterSet.newlines)
+        
+        // Iterate over the lines
+        var eventLines = [String]()
+        for line in lines {
+            if line == EventKeys.eventBegin {
+                inEvent = true
+            } else if line == EventKeys.eventEnd {
+                inEvent = false
+                let parsedLine = parse(lines: eventLines)
+                let event = Event(with: parsedLine)
+                events.append(event)
+                eventLines.removeAll()
+            } else if inEvent {
+                eventLines.append(line)
+            }
+        }
+        return Calendar(events: events)
+    }
+    
+    /// Parse ICS file from URL
+    public static func parse(url: URL, completion: @escaping (Result<Calendar,ParseError>) -> Void) {
```

Comment:
Where do you use this? why do we have that?

Reply:
Removed.